hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -335,10 +335,10 @@ def extract_lambda_source(f):
                 break
             except SyntaxError:
                 continue
-    if tree is None and source.startswith("@"):
-        # This will always eventually find a valid expression because
-        # the decorator must be a valid Python function call, so will
-        # eventually be syntactically valid and break out of the loop.
+    if tree is None and source and source[0] in ["@", "."]:
```

Comment:
Funnily enough, this breaks 
https://github.com/HypothesisWorks/hypothesis/blob/88f80e8cacfad6e1f7e076b3ad0a410f924156a7/hypothesis-python/tests/cover/test_custom_reprs.py#L57

-- which was changed as part of the now-dropped "fluff", but not anymore.

The problem is that the lambda is no longer just `"<unknown>"`, so the lazy-strategy `force_repr` fails at construction time. Not sure at which level this should be fixed?

Reply:
I think the correct fix is to not force a repr if attempting to calculate that raises an error - the laziness is more important.  That will be an additional bugfix, since the test was only coincidentally passing due to this problem.

style note: you can pass a tuple of possible prefixes, so `source.startswith(("@", "."))` works.

Reply:
Yep, agreed, we could even warn in that case ;-)

Anyway, I wrapped a few `force_repr`s in try-except; only the `filter` one was needed to pass the test - not the lazy one as I thought, but I wrapped that too.